<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<HTML>
<head>
<!-- official OU colors:  crimson #4F1315 cream #FDF908  source: https://alumni.ou.edu/home/webcomm/webguide/designconsiderations.html -->
<!-- off white #f5f4f0   reddish slate #E4D2C1  OU brickred #4F1315   light green #FFEFB2  light blue #F6F8FF   light-green #fceeaf   royal blue #0D2A70   light-green #FFFCCC  light gray F6F4F0 -->
<!--<body bgcolor=#FDF9D8  style="font-family:arial;font-size:15px"> -->
<link rel="stylesheet" href="style.css">
<style type="screen.css">
    <style>
      a {
        text-decoration: none;
      }
      a:link {
        color: #6b5b95;
      }
      a:visited {
        color: #feb236;
      }
      a:hover {
        color: #d64161;
      }
</style>

<script type="text/javascript" async
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
</script>


<!-- Format code blocks. Must use code tag inside of pre tags to get the block effect-->
<style type="text/css">
	pre code {
	  background-color: #eee;
	  border: 1px solid #999;
	  display: block;
	  padding: 20px;
	}
</style>


<!-- Format the vertical spacing in lists. 	    border:1px solid grey;-->
<style type="text/css">
	li{
	    margin-top: 10px;
	}

	li:first-child {
	    margin-top:0;
	}
</style>

<!-- Format table -->

<style>
<!--
table.table1 {
  border:1px solid #0F2080;
  width:100%;
  table-layout:fixed;
}

table.table1 caption {
	font-family: Arial, Helvetica, sans-serif;
    text-align: center;
    margin-bottom: 5px;
    font-size: 240%;
    padding: 5px;
    font-weight: bold;
}


table.table1 tbody>tr>:nth-child(1){
 color:#0F2080;
 text-align:center;
 font-size: 2em;
 width: 40;
}

table.table1 tbody>tr>:nth-child(2){
 text-align:center;
 width: 200;
}

table.table1 tbody>tr>:nth-child(3){
 color:#0F2080;
 text-align:left;
 padding: 10px;
 font-size: 1.2em;
 word-wrap:break-word;
 width: 200;
}

table.table1 th { 
	font-size: 1.4em;
	border:1px solid  #0F2080; 
	text-align:center;
}

table.table1 td{
	border:1px solid;
	color: #0F2080; 
}



table.table2 {
    border:1px solid black;
    width:100%;
    table-layout:fixed;
    font-family: arial, sans-serif;
    border-collapse: collapse;
    width: 100%;
}

table.table2 tbody>tr>:nth-child(1){
 color:black;
 text-align:left;
 font-size: 1em;
 border: 1px solid black;
 padding: 8px;
 width: 80px;
}

table.table2 tbody>tr>:nth-child(2){
    color:black;
    text-align:left;
    font-size: 1em;
    border: 1px solid black;
    padding: 8px;
    width: 300px;
}

table.table2 th { 
	font-size: 1.0em;
	border:1px solid black; 
	text-align:left;
}

table.table2 tr:nth-child(even) {
  background-color: #dddddd;
}

-->
</style>


<!--
table.table2 td, th {
  border: 1px solid  #A95AA1;
  text-align: left;
  padding: 8px;
}
-->



<style type="text/css">
<!--
 .tab { margin-left: 25px; }
-->
</style>


<!-- tab a section -->
<STYLE TYPE="text/css">
<!--
.indented
 {
 padding-left: 50pt;
 padding-right: 50pt;
 }
-->
</STYLE>

<!-- tab a section -->
<STYLE TYPE="text/css">

ul {
  list-style-position: inside;
  padding-left: 0;
}
</STYLE>

<!-- A nice ligth grey: bgcolor=#f5f4f0  -->
</head>
<body style="font-family:Arial, Helvetica, sans-serif; font-size:15px">

<link rel="stylesheet" href="style.css">

<h1> <em>jupyterlabpymolpysnips:</em> PyMOL Python Script Writing with Code Templates in JupyterLab</h1>

This webpage is associated with the GitHub repository for the <a href="https://github.com/MooersLab/jupyterlabpymolpysnips">jupyterlabpymolpysnips library</a> of snippets of Python Code for the molecular graphics program PyMOL.

<a name='categories'> <h3>Categories of snippets:</h3> </a>

<ol>
<li><a href="#AlternateLocators"> Alternate Locators</a></li>
<li><a href="#ChangeOrientation"> Change Orientation</a></li>
<li><a href="#ColorScheme">Color Scheme</a></li>
<li><a href="#ElectronDensity">Electron Density</a></li>
<li><a href="#FileIO">FileIO</a></li>
<li><a href="#H-bonds">H-bonds</a></li>
<li><a href="#Help">Help</a></li>
<li><a href="#Jupyter">Jupyter</a></li>
<li><a href="#Labels">Labels</a></li>
<li><a href="#ListSnippets">List Snippets</a></li>
<li><a href="#Measurements">Measurements</a></li>
<li><a href="#MolecularRepresentation">Molecular Representations</a></li>
<li><a href="#NucleicAcids">Nucleic Acids</a></li>
<li><a href="#Objects">Objects</a></li>
<li><a href="#Print">Print</a></li>
<li><a href="#Programming">Programming</a></li>
<li><a href="#Pymolrc">Pymolrc</a></li>
<li><a href="#Selection">Selection</a></li>
<li><a href="#Settings">Settings</a></li>
<li><a href="#Stereo">Stereo</a></li>
<li><a href="#Trajectories">Trajectories</a></li>
<li><a href="#UnitCellDisplay">Unit Cell Display</a></li>
<li><a href="#Workshop">Workshop</a></li>
</ol>



<a name='AlternateLocators'> <h3>Alternate Locators:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>hidealtloc   </td><td>Hide the partially occupied atoms with the part b alternate locator.</td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='ChangeOrientation'> <h3>Change Orientation:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>rotate       </td><td>Rotate a selection about and axis by a given angle.</td></tr>
<tr><td>rv           </td><td>Return settings in rounded format.                 </td></tr>
<tr><td>turnAboutAxis</td><td>Turn about axis.                                   </td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='ColorScheme'> <h3>Color Scheme:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>cblind               </td><td>Apply color blind friendly to ribbon diagrams. Edit the path to the Pymol-script-repo in your computer account. See PyMOL wiki for more information about the Pymol-script-reo.                                                                                                                                                                                                   </td></tr>
<tr><td>cribbon              </td><td>Color ribbon H red, strand yellow, loop green.                                                                                                                                                                                                                                                                                                                                    </td></tr>
<tr><td>grayscale            </td><td>Apply grayscale coloring using a grayscale version of the PyMOL colors for the elements. This is a Python function. It is invoked in a script file via grayscale(). There is a corresponding gscale shortcut in pymolshortcuts.py that is invoked in a pml script by entering gsale if the functions in pymolshortcuts.py have been loaded with the run pymolshortcuts.py command.</td></tr>
<tr><td>printColorByAtomCodes</td><td>Print the codes for color-by-atom (util.cba*) alternates.                                                                                                                                                                                                                                                                                                                         </td></tr>
<tr><td>volumeRamp           </td><td>Volume ramp.                                                                                                                                                                                                                                                                                                                                                                      </td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='ElectronDensity'> <h3>Electron Density:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>carvedIsomesh       </td><td>Carved isomesh representation of electron density.                                                                            </td></tr>
<tr><td>carvedIsosurface    </td><td>Carved isosurface representation of electron density.                                                                         </td></tr>
<tr><td>carvedVolume        </td><td>Carved volume representation of electron density.                                                                             </td></tr>
<tr><td>fetch2FoFcIsomesh   </td><td>Fetch 2FoFc map as an isomesh.                                                                                                </td></tr>
<tr><td>fetch2FoFcIsosurface</td><td>Fetch 2FoFc map as an isosurface. Edit the PDB-ID code. Use lowercase letter for the fifth character to select a single chain.</td></tr>
<tr><td>fetch2FoFcVolume    </td><td>Fetch 2FoFc map as a volume.                                                                                                  </td></tr>
<tr><td>fetchThreeMaps      </td><td>Display three electron density maps as isomesh.                                                                               </td></tr>
<tr><td>loadThreeMaps       </td><td>Three electron density as Isomesh.                                                                                            </td></tr>
<tr><td>threeMapsIsosurface </td><td>Display three electron density maps as isosurfaces.                                                                           </td></tr>
<tr><td>threeMapsVolume     </td><td>Three electron density as volumes.                                                                                            </td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='FileIO'> <h3>FileIO:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>cblindCartoon</td><td>Color cartoon with colorblind friendly colors. Requires that the pymolshortcuts.py file is loaded. This has been applied to PDB-ID 7JU6. The protein is human RET kinase, and the drug is selpercatinib, a FDA approved drug for treating several cancers.</td></tr>
<tr><td>fetchCIF     </td><td>Fetch the atomic coordinates as a cif file from the PDB.                                                                                                                                                                                                  </td></tr>
<tr><td>fetchFoFc    </td><td>Fetch fofc map from the PDB.                                                                                                                                                                                                                              </td></tr>
<tr><td>loadPDBfile  </td><td>Load a pdb file in the current directory.                                                                                                                                                                                                                 </td></tr>
<tr><td>printPath    </td><td>Print the path to the currently used PyMOL binary.                                                                                                                                                                                                        </td></tr>
<tr><td>saln         </td><td>Save an aln flle with a timestamp.                                                                                                                                                                                                                        </td></tr>
<tr><td>salnpy       </td><td>Save an aln flle with a timestamp. Python version.                                                                                                                                                                                                        </td></tr>
<tr><td>savePNG      </td><td># Save a png file of current scene to the current directory. PyMOL writes out only png files. This file may need to be converted to a tiff file. See the png2tiff snippet for a bash script that converts all png files in a folder into tiff files.  # 1: png filename  # 2: x-dimension in pixels. # 3: y-dimension in pixels, 1600 x 1000 approximates the golden ratio. Usually want a square for multipanel figures.  # 4: dots per inch,  # 5: ray tracing off, 0; ray tracing on, 1 # should also consider image without ray tracing shadows.  </td></tr>
<tr><td>sccp4        </td><td>Save electron density map flle with timestamp.                                                                                                                                                                                                            </td></tr>
<tr><td>sccp4py      </td><td>Save electron density map flle with timestamp.  </td></tr>
<tr><td>sdae         </td><td>Save dae flle with timestamp.  </td></tr>
<tr><td>sdaepy       </td><td>Save dae flle with timestamp.  </td></tr>
<tr><td>spng         </td><td>Save png flle with timestamp. </td></tr>
<tr><td>spngpy       </td><td>Save png flle with timestamp. </td></tr>
<tr><td>spse         </td><td>Save pse flle with timestamp    </td></tr>
<tr><td>wallart      </td><td>Reset hash_max from 100 to 2000 to enable the saving of 28" by 28".    </td></tr>
<tr><td>wallartpy    </td><td>Reset hash_max from 100 to 2000 to enable the saving of 28" by 28".    </td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='H-bonds'> <h3>H-bonds:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>distance     </td><td>H-bond distance between a H-bond donor and acceptor.</td></tr>
<tr><td>drawHbonds   </td><td>Display H-bonds as dashes colored black.            </td></tr>
<tr><td>hbonddash    </td><td>Set up H-bond dashes.                               </td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='Help'> <h3>Help:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>cmddocs                   </td><td>Print in the command history window the docstrings of all of the functions in the cmd module.                                                                        </td></tr>
<tr><td>github                    </td><td>Print url of README.md file of the pymolsnips repository.                                                                                                            </td></tr>
<tr><td>helpDocs                  </td><td>Return the docustring for the help submodule. This command is more concise:  help help.                                                                              </td></tr>
<tr><td>presetDocs                </td><td>Return the docustring for the preset submodule. The command help preset fails to return anything. The command help(pymol.preset) has the same effect as help(preset).</td></tr>
<tr><td>printDoc                  </td><td>Print document string of a function.                                                                                                                                 </td></tr>
<tr><td>printDocpy                </td><td>Print document string of a function.                                                                                                                                 </td></tr>
<tr><td>pymoldocs                 </td><td>Return to the command history window the docstrings for all of the functions in a module.                                                                            </td></tr>
<tr><td>pymoldocspy               </td><td>Return to the command history window the docstrings for all of the functions in a module.                                                                            </td></tr>
<tr><td>writeCommandReference2HTML</td><td>Write the command reference to html file in the present working directory.                                                                                           </td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='Jupyter'> <h3>Jupyter:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>AO                     </td><td>Run the AO function from the pymolshortcuts.py file to generate the photorealistic effect.                                                                                                                        </td></tr>
<tr><td>AOBW                   </td><td>Run the AOBW function from the pymolshortcuts.py file to generate photorealistic effect in grayscale.                                                                                                             </td></tr>
<tr><td>AOD                    </td><td>Run the AOD function from the pymolshortcuts.py file to generate photorealistic effect with carbons colored black.                                                                                                </td></tr>
<tr><td>AODBW                  </td><td>Run the AODBW function from the pymolshortcuts.py file to generate photorealistic effect with carbons colored black and all other atoms colored in grayscale.                                                     </td></tr>
<tr><td>PE125                  </td><td>Run the PE125 function from the pymolshortcuts.py file to show the pearl effect with the inner sphere scaled by 0.125 of the van der Waals surface.                                                               </td></tr>
<tr><td>PE25                   </td><td>Run the PE25 function from the pymolshortcuts.py file to show the pearl effect with the inner sphere scaled by 0.25 of the van der Waals surface.                                                                 </td></tr>
<tr><td>PE33                   </td><td>Run the PE33 function from the pymolshortcuts.py file to show the pearl effect with the inner sphere scaled by 0.33 of the van der Waals surface.                                                                 </td></tr>
<tr><td>PE50                   </td><td>Run the PE50 function from the pymolshortcuts.py file to show the pearl effect with the inner sphere scaled by 0.50 of the van der Waals surface.                                                                 </td></tr>
<tr><td>PE66                   </td><td>Run the PE66 function from the pymolshortcuts.py file to show the pearl effect with the inner sphere scaled by 0.66 of the van der Waals surface.                                                                 </td></tr>
<tr><td>PE75                   </td><td>Run the PE75 function from the pymolshortcuts.py file to show the pearl effect with the inner sphere scaled by 0.75 of the van der Waals surface.                                                                 </td></tr>
<tr><td>PE85                   </td><td>Run the PE85 function from the pymolshortcuts.py file to show the pearl effect with the inner sphere scaled by 0.85 of the van der Waals surface.                                                                 </td></tr>
<tr><td>cav                    </td><td>Run the cav function from the pymolshortcuts.py file to show buried cavities and pockets as molecular surfaces.                                                                                                   </td></tr>
<tr><td>cntccp4emaps           </td><td>Count number of *.ccp4 (electron density map) files in current directory.                                                                                                                                         </td></tr>
<tr><td>cntfiles               </td><td>Count number of files in current directory.                                                                                                                                                                       </td></tr>
<tr><td>cntlogs                </td><td>Count number of *.log files in current directory.                                                                                                                                                                 </td></tr>
<tr><td>cntmtzs                </td><td>Count number of *.mtz files in current directory.                                                                                                                                                                 </td></tr>
<tr><td>cntpdbs                </td><td>Count number of *.pdb files in current directory.                                                                                                                                                                 </td></tr>
<tr><td>cntpmls                </td><td>Count number of *.pml files in current directory.                                                                                                                                                                 </td></tr>
<tr><td>cntpngs                </td><td>Count number of *.png files in current directory.                                                                                                                                                                 </td></tr>
<tr><td>cntpses                </td><td>Count number of *.pse files in current directory.                                                                                                                                                                 </td></tr>
<tr><td>colorh1                </td><td>Run the colorh1 function from the pymolshortcuts.py file to color protein molecules according to the Eisenberg hydrophobicity scale, scheme 1.                                                                    </td></tr>
<tr><td>colorh2                </td><td>Run the colorh2 function from the pymolshortcuts.py file to color protein molecules according to the Eisenberg hydrophobicity scale, scheme 2.                                                                    </td></tr>
<tr><td>hb                     </td><td>Creates an object of all H-bonds found by PyMOL.                                                                                                                                                                  </td></tr>
<tr><td>importIPythonDisplay   </td><td>Imports for using IPython to display images loaded from disk in notebook cells.                                                                                                                                   </td></tr>
<tr><td>importPyMOLandShortcuts</td><td>Imports needed for most uses of pymol in Jupyter. Combination of importPyMOL and importPythonDisplay.                                                                                                             </td></tr>
<tr><td>importPyMOLcmd         </td><td>Import the cmd class from the pymol api.                                                                                                                                                                          </td></tr>
<tr><td>importShortcuts        </td><td>Import for loading the functions in the pymolshortcuts.py file. These functions can be run inside cmd.do() without the trailing (). For example, cmd.do('rv').                                                    </td></tr>
<tr><td>imports4PyMOLjupyter   </td><td>Imports needed for most uses of pymol in Jupyter. Combination of importPyMOL and importPythonDisplay.                                                                                                             </td></tr>
<tr><td>ipymolProtein          </td><td>Demo of the use of the RPC server with a protein via ipymol.                                                                                                                                                      </td></tr>
<tr><td>ipymolStart            </td><td>Code to start the RPC server with ipymol. Start pymol in terminal with pymol -R; select pymol.python as the kernel in juptyer. You may have to create this kernel for the Python interpreter that is inside PyMOL.</td></tr>
<tr><td>kernel                 </td><td>A kernel.json file for runnig PyMOL python interpreter in the Jupyter notebook. This code should reside in a folder named pymol.python in the ~/Library/Jupyter/kernels.                                          </td></tr>
<tr><td>loadImage             </td></tr>
<tr><td>nmr                    </td><td>Show all models in a nmr structure.                                                                                                                                                                               </td></tr>
<tr><td>nmroff                 </td><td>Hide all but first model in a nmr structure.                                                                                                                                                                      </td></tr>
<tr><td>nmroffpy               </td><td>Hide all but the first model in a nmr structure.                                                                                                                                                                  </td></tr>
<tr><td>nmrpy                  </td><td>Show all models in a nmr structure.                                                                                                                                                                               </td></tr>
<tr><td>rdkrpcChem             </td><td>Demo of the use of the RPC server with a drug compound via the rdkit python module.                                                                                                                               </td></tr>
<tr><td>rdkrpcProtein          </td><td>Demo of the use of the RPC server with a protein via rdkit.                                                                                                                                                       </td></tr>
<tr><td>rmd                    </td><td>Remove all measurement objects in the interal GUI.                                                                                                                                                                </td></tr>
<tr><td>rmhb                   </td><td>Delete all H-bonds in the selection, which is all by default.                                                                                                                                                     </td></tr>
<tr><td>rv                     </td><td>Run the rv function from the pymolshortcuts.py file.                                                                                                                                                              </td></tr>
<tr><td>rvi                    </td><td>Return settings in rounded format while running PyMOL via the RCP server ipymol in a jupyter notebook. This is a modified version of the roundview.py script.                                                     </td></tr>
<tr><td>rvr                    </td><td>Return settings in rounded format while running PyMOL via the RCP server rdkit in a jupyter notebook. This is a modified version of the roundview.py script.                                                      </td></tr>
<tr><td>timcolor               </td><td>Run the timcolor function from the pymolshortcuts.py file to color atoms accordings to Tim Mather's biophysical coloring scheme for proteins.                                                                     </td></tr>
<tr><td>yrb                    </td><td>Run the yrb function from the pymolshortcuts.py file.                                                                                                                                                             </td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='Labels'> <h3>Labels:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>centerpi      </td><td>Center pi. Edit the atoms selected for positioning the pseudoatom.                                                                                                                                   </td></tr>
<tr><td>displayFonts  </td><td>Print to the screen as labels the 21 font ids in their corresponding fonts in a grid. Each label is an object and appears in the internal gui. You can turn on and off the display of specific fonts.</td></tr>
<tr><td>displayFontspy</td><td>Print to the screen as labels the 21 font ids in their corresponding fonts in a grid. Each label is an object and appears in the internal gui. You can turn on and off the display of specific fonts.</td></tr>
<tr><td>labelCAs      </td><td>Label the CA atoms with the Ala333 style format.                                                                                                                                                     </td></tr>
<tr><td>labelMainChain</td><td>Label the main chain atoms with the following: resn,resi,atom name.                                                                                                                                  </td></tr>
<tr><td>labelResnResi </td><td>Label CA atom with single-letter residue name and residue number.                                                                                                                                    </td></tr>
<tr><td>labelSS       </td><td>Label SS.                                                                                                                                                                                            </td></tr>
<tr><td>labelWatersHOH</td><td>Label waters with HOH and their residue number.                                                                                                                                                      </td></tr>
<tr><td>labelWatersW  </td><td>Label waters with W and their reisude number.                                                                                                                                                        </td></tr>
<tr><td>oneLetter     </td><td>Switch from three letter code to one-letter code for amino acids.                                                                                                                                    </td></tr>
<tr><td>pseudolabel   </td><td>Position label with pseudoatom.                                                                                                                                                                      </td></tr>
<tr><td>sigang        </td><td>Set angle labels to display 2 decimals places.                                                                                                                                                       </td></tr>
<tr><td>sigdihedral   </td><td>Set dihedral labels to display 2 decimals places to the right of the decimal point.                                                                                                                  </td></tr>
<tr><td>sigdist       </td><td>Set distance labels to display 2 decimals.                                                                                                                                                           </td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='ListSnippets'> <h3>List Snippets:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>lsSnips      </td><td>List all snips by Snippet Name and description.</td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='Measurements'> <h3>Measurements:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>listLigandProteinDistances</td><td>Print a list of protein--ligand distances. Code by Dan Kulp. Updated for Python3.</td></tr>
<tr><td>ms                        </td><td>Measure surface area of the selection with the msms_pymol.py script.             </td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='MolecularRepresentation'> <h3>Molecular Representations:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>ao                            </td><td>Apply the ambient occlussion effect to get the photorealistic effect.                                                                                                                                                                                                                                                                                                          </td></tr>
<tr><td>ao                            </td><td>Apply the ambient occlussion effect to get the photorealistic effect.                                                                                                                                                                                                                                                                                                          </td></tr>
<tr><td>aobw                          </td><td>Ambient occlussion in grayscale.                                                                                                                                                                                                                                                                                                                                               </td></tr>
<tr><td>aod                           </td><td>Ambient occlussion with carbon atoms colored black.                                                                                                                                                                                                                                                                                                                            </td></tr>
<tr><td>aodbw                         </td><td>Ambient occlussion in grayscale with carbon atoms colored black. Note: requires the gscale() function from pymolshortcuts.py. Download this script from http://GitHub.com/MooersLab/pymolshortcuts. Load the functions from this script with the command "cmd.do("run pymolshortcuts.py").                                                                                     </td></tr>
<tr><td>bs                            </td><td>Ball and stick representation.                                                                                                                                                                                                                                                                                                                                                 </td></tr>
<tr><td>bsfr                          </td><td>Ball-and-stick plus filled ring representation for ligands.                                                                                                                                                                                                                                                                                                                    </td></tr>
<tr><td>bu                            </td><td>Generate the biological unit using the quat.py script.                                                                                                                                                                                                                                                                                                                         </td></tr>
<tr><td>coordinate                    </td><td>Coordinate covalent bonds to metals and H-bonds from RNA.                                                                                                                                                                                                                                                                                                                      </td></tr>
<tr><td>cspheres                      </td><td>Colored spheres.                                                                                                                                                                                                                                                                                                                                                               </td></tr>
<tr><td>discreteCartoonColoring       </td><td>Turn on discrete colors between secondary structure elements.                                                                                                                                                                                                                                                                                                                  </td></tr>
<tr><td>doubleBond                    </td><td>Valence bond.                                                                                                                                                                                                                                                                                                                                                                  </td></tr>
<tr><td>drawLinks                     </td><td>Connect the alpha carbons of residue 1 with 10, 6 with 16, 7  with 17 and 8 with 18. Note that this example requires the draw_links.py [http://pldserver1.biochem.queensu.ca/~rlc/work/pymol/draw_links.py] by Robert Campbell.                                                                                                                                                </td></tr>
<tr><td>ellipcol                      </td><td>Set color of thernal ellipsoids. The PDB must have anisotopic temperature factors. See https://pymolwiki.org/index.php/Color_Values for the PyMOL colors.                                                                                                                                                                                                                      </td></tr>
<tr><td>filledRing                    </td><td>Filled rings in nucleic acids.                                                                                                                                                                                                                                                                                                                                                 </td></tr>
<tr><td>fog                           </td><td>Blur the background atoms.                                                                                                                                                                                                                                                                                                                                                     </td></tr>
<tr><td>grayscalepy                   </td><td>Apply grayscale coloring using a grayscale version of the PyMOL colors for the elements. This is a Python function. It is invoked in a script file via gscale(). There is a corresponding gscale shortcut in pymolshortcuts.py that is invoked in a pml script by entering gsale if the functions in pymolshortcuts.py have been loaded with the run pymolshortcuts.py command.</td></tr>
<tr><td>his31asp70                    </td><td>Display the famous Asp70-His31 salt-bridge from T4 lysozyme that contributes3-5 kcal/mol to protein stability.                                                                                                                                                                                                                                                                 </td></tr>
<tr><td>loadPDBbs                     </td><td>Load PDB ball-and-stick.                                                                                                                                                                                                                                                                                                                                                       </td></tr>
<tr><td>loadPDBnb                     </td><td>Load PDB nb spheres.                                                                                                                                                                                                                                                                                                                                                           </td></tr>
<tr><td>molscriptRibbon               </td><td>Show cartoon in the style of Molscript ribbons.                                                                                                                                                                                                                                                                                                                                </td></tr>
<tr><td>oneBondThicknessColor         </td><td>To change stick color and radius for the bond between atom 2 and 3, use the set_bond command.                                                                                                                                                                                                                                                                                  </td></tr>
<tr><td>pearl                         </td><td>The pearl effect is made with two spheres with the outer sphere being transparent.                                                                                                                                                                                                                                                                                             </td></tr>
<tr><td>puttyCartoon                  </td><td>Create a putty cartoon.                                                                                                                                                                                                                                                                                                                                                        </td></tr>
<tr><td>ringMode                      </td><td>Set the ring mode to a value between 0  and 6 in cartoons of nucleic acids.                                                                                                                                                                                                                                                                                                    </td></tr>
<tr><td>rmwater                       </td><td>Remove waters from molecular object.                                                                                                                                                                                                                                                                                                                                           </td></tr>
<tr><td>sas                           </td><td>Show the solvent excluded surface.                                                                                                                                                                                                                                                                                                                                             </td></tr>
<tr><td>saxsEnvelope                  </td><td>Display SAXS envelope. Edit to enter the name of the bead model object.                                                                                                                                                                                                                                                                                                        </td></tr>
<tr><td>scaleRadiusColor              </td><td>Scale the radius and color of atoms as spheres by property in the B-value column.                                                                                                                                                                                                                                                                                              </td></tr>
<tr><td>scaleRadiusColorPythonInsertpy</td><td>Python block insert for scaleRadiusColorpy.                                                                                                                                                                                                                                                                                                                                    </td></tr>
<tr><td>scaleRadiusColorpy            </td><td>Scale the radius and color of atoms as spheres by property in the B-value column.                                                                                                                                                                                                                                                                                              </td></tr>
<tr><td>setLigandValenceOn            </td><td>Display the bond valence of ligands only.                                                                                                                                                                                                                                                                                                                                      </td></tr>
<tr><td>setcolor                      </td><td>Set color name to a RGB code.                                                                                                                                                                                                                                                                                                                                                  </td></tr>
<tr><td>sidehChainHelper              </td><td>In cartoons, hide the backbone atoms of selected residues when showing then as sticks.                                                                                                                                                                                                                                                                                         </td></tr>
<tr><td>solventRadius                 </td><td>Set radius of ball used to make solvent accessible surface.                                                                                                                                                                                                                                                                                                                    </td></tr>
<tr><td>stack                         </td><td>Base-stacking figure.                                                                                                                                                                                                                                                                                                                                                          </td></tr>
<tr><td>stateOne                      </td><td>Select state 1 from a model with multiple states.                                                                                                                                                                                                                                                                                                                              </td></tr>
<tr><td>waterTriple                   </td><td>Examples of a triple water pentagon. Zoom in on the selection. Edit by changing the residue number.                                                                                                                                                                                                                                                                            </td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='NucleicAcids'> <h3>Nucleic Acids:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>basePairStacking                        </td><td>This code make as standard base stacking diagram with ball and stick representation.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         </td></tr>
<tr><td>brokenNucleicBackbone                   </td><td>Create bonds between phosphorous and O3* atoms in a low-resolution DNA structure that is 80 base pairs long. Edit the selections below, which are ranges of residue numbers and edit the molecular object name (5fur).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       </td></tr>
<tr><td>dssrBlock1                              </td><td>Combining DSSR block representation with regular PyMOL cartoons after loading the dssr_block.py script by Thomas Holder.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     </td></tr>
<tr><td>dssrBlock2                              </td><td>DSSR block representation with fused blocks after loading the dssr_block.py script by Thomas Holder. The x3dna-dssr executable needs to be in the PATH.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      </td></tr>
<tr><td>dssrBlock3                              </td><td>DSSR block representation for a multi-state example after loading the dssr_block.py script by Thomas Holder. The x3dna-dssr executable needs to be in the PATH. Edit the path to Thomas Holder's block script.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               </td></tr>
<tr><td>dssrBlock4                              </td><td>DSSR block representation with custom coloring after loading the dssr_block.py script by Thomas Holder. The x3dna-dssr executable needs to be in the PATH.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   </td></tr>
<tr><td>nucleicAcidBackboneTubesSticks          </td><td>This code shows the cartoon backbone tube as 65% transparent. It hides the rungs of the cartoon. It shows all of the non-H atoms are sticks. The motivation is to have the cartoon highlight the backbone without dominanting the scene.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     </td></tr>
<tr><td>nucleicAcidCartoon                      </td><td>Settings for nucliec acid cartoon. The dark blue used for electron density maps is called `density`. The cartoon_ladder_radius should be renamed the cartoon_rung_radius. The dimensions are in Angstroms.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   </td></tr>
<tr><td>nucleicAcidCartoon2Strands              </td><td>Coloring two strand differently of a double helix makes it easier to for the viewer to distinguish the two strands. The set command has the syntax of setting_name [, setting_value [, selection [,state ]]] . In this case, the selection has to be global, object, object-state, or per-atom settings. It cannot be a named selection. This is a weak spot in PyMOL. Coloring two strand differently of a double helix makes it easier to for the viewer to distinguish the two strands. Many double-stranded helices have one strand in the asymmetric unit. The second strand is in the biological unit. The coordinates for the second strand are in the pdb1 file type at the PDB. The second strand is in the second state, which is equivalent to the second model in the pdb file. The strands are labeled chain A and B (via the cartoon_nucliec_acid_color setting). The bases are colored differently too (via the cartoon_ladder_color setting).</td></tr>
<tr><td>nucleicAcidCartoonFilledRings           </td><td>The code provides a cartoon of the loaded nucleic acid that has the ladder rungs replaced by filled rings that are colored by atom type. The code can be applied to any nucleic acid. It is derived from the FR shortuct in pymolshortcuts.py.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               </td></tr>
<tr><td>nucleicAcidColorbySequence              </td><td>This code colors the nucleotides by base seqence. It can be applied to any nucleic acid.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     </td></tr>
<tr><td>nucleicAcidDumbellCartoonColorbySequence</td><td>This code colors by the nucleotides by base seqence. The backcone is shown as a flatten ribbon with rolled edges that give the dumbell effect. The code can be applied to any nucleic acid. The code is dervied from the CR and DU shortcuts.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                </td></tr>
<tr><td>nucleicAcidFlatRibbonColorbySequence    </td><td>This code colors the nucleotides by base seqence. It can be applied to any nucleic acid. It is dervied from the CR shortcut. The backcone is shown as a flatten ribbon.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      </td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='Objects'> <h3>Objects:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>allPairs    </td><td>This is a two-fold nested list comprehension for any all-parwise operation on the currently loaded objects. Replace optAlginRNA with object from any other function that operations on a pair of structrures.</td></tr>
<tr><td>listObjects           </td><td>Create a list of objects in the internal gui and print this list to the screen.                                                                                                                              </td></tr>
<tr><td>loadAndAlignManyFiles1</td><td>These are the instructions for loading and aligning multiple files. To save multiple models in a file to separate pdb files.                                                                                 </td></tr>
<tr><td>loadAndAlignManyFiles2</td><td>To align all of the loaded RNA structures in all possible combinations by their C1' carbon atoms.                                                                                                            </td></tr>
<tr><td>loadAndAlignManyFiles3</td><td>These are the instructions for loading and aligning multiple files.                                                                                                                                          </td></tr>
<tr><td>loadManyFiles         </td><td>Load into PyMOL multiple files with a common file stem. The is a script by Robert Campbell that has been updated for Python3.                                                                                </td></tr>
<tr><td>optAlignRNA           </td><td>OptiAlign.py by Jason Vertree modified for aligning multiple RNA structures.                                                                                                                                 </td></tr>
<tr><td>optAlignRNA           </td><td>OptiAlign.py by Jason Vertree modified for aligning multiple RNA structures.                                                                                                                                 </td></tr>
<tr><td>saveSeppy             </td><td>Saves multiple objects into multiple files using an optional prefix name.                                                                                                                                    </td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='Print'> <h3>Print:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>aveB4resiX                 </td><td>Prints the residue number and the average bfactor. Uses reduce and lambda, builtin Python functional porgramming functions. Note that you need to convert the length of the list of Bfactors from an integer to a float before division into the sum.</td></tr>
<tr><td>aveB4resiXpy               </td><td>AveBResiX, prints the residue number and the average bfactor. Uses reduce and lambda, builtin Python functional porgramming functions. Note that you need to convert the length of the list of Bfactors from an integer.                             </td></tr>
<tr><td>averageB                   </td><td>Average the B-factors by using a regular list as opposed to a stored list in PyMOL. Edit the selection as needed.                                                                                                                                    </td></tr>
<tr><td>fasta                      </td><td>Print Fasta from PDB file.                                                                                                                                                                                                                           </td></tr>
<tr><td>fastapy                    </td><td>Python version of the command to print the sequence from a PDB file in the fasta format.                                                                                                                                                             </td></tr>
<tr><td>findHbonds                 </td><td>Find H-bonds around a residue.                                                                                                                                                                                                                       </td></tr>
<tr><td>getCoordinates             </td><td>Get coordinates.                                                                                                                                                                                                                                     </td></tr>
<tr><td>getCoordinatespy           </td><td>Python version of getCoordinates snippets. Note that the python2 print statement stills works  in pml scripts.                                                                                                                                       </td></tr>
<tr><td>numResiNucleic             </td><td>Print the number of residues in a nulceic acid (all chains).                                                                                                                                                                                         </td></tr>
<tr><td>numResiNucleicChainA       </td><td>Print the number of residues in a nulceic acid chain.                                                                                                                                                                                                </td></tr>
<tr><td>numResiProtein             </td><td>Print the number of residues in a protein.                                                                                                                                                                                                           </td></tr>
<tr><td>numResiProteinChainA       </td><td>Print the number of residues in chain A of a protein.                                                                                                                                                                                                </td></tr>
<tr><td>printBs                    </td><td>Print the B-factors of a residue.                                                                                                                                                                                                                    </td></tr>
<tr><td>printBs2digits             </td><td>Print B--factors for a residue with the B-factors rounded off to two decimal places.                                                                                                                                                                 </td></tr>
<tr><td>printBspartB               </td><td>Print B factors of part B of a residue.                                                                                                                                                                                                              </td></tr>
<tr><td>printNameB4ResiX           </td><td>Print name and b-factor for a residue.                                                                                                                                                                                                               </td></tr>
<tr><td>printPathpy                </td><td>Print the path to the currently used PyMOL binary.                                                                                                                                                                                                   </td></tr>
<tr><td>printResiResnNameB4ResiX   </td><td>Print resn, resi, atom name, and b-factor.                                                                                                                                                                                                           </td></tr>
<tr><td>printResiResnNameB4ResiXNoH</td><td>Print name and b-factor for a residue or residue range (e.g. 81:120). The noH variant.                                                                                                                                                               </td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='Programming'> <h3>Programming:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>emacsjupyterSourceBlock</td><td>Source block template in org-mode with emacs-jupyter package.                                              </td></tr>
<tr><td>obipythonSourceBlock   </td><td>Source block template in org-mode with the ob-ipython package.                                             </td></tr>
<tr><td>printAtomNames         </td><td>Print the atom names of a residue.                                                                         </td></tr>
<tr><td>printAtomNumbers       </td><td>Print the atom number2 of a residue.                                                                       </td></tr>
<tr><td>printBfactors          </td><td>Print the bfactors of a residue.                                                                           </td></tr>
<tr><td>printCoordinates       </td><td>Print the coordinates of the atoms in a residue.                                                           </td></tr>
<tr><td>printNamesCoordinates  </td><td>Print the atom names and coordinates of the atoms in a residue.                                            </td></tr>
<tr><td>printNamesCoordinates  </td><td>Print the atom names as tuples and coordinates of the atoms in a residue as a list.                        </td></tr>
<tr><td>printVDWradii          </td><td>Print the van der Waals radii of the atoms in of a residue.                                                </td></tr>
<tr><td>renameChain            </td><td>Rename a chain.                                                                                            </td></tr>
<tr><td>renumAtoms             </td><td>Add or substract a atom number offset.                                                                     </td></tr>
<tr><td>renumResi              </td><td>Add or substract a residue number offset.                                                                  </td></tr>
<tr><td>synch                  </td><td>Wait unitl all current commands have been executed. A timeout ensures that that command ecentually returns.</td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='Pymolrc'> <h3>Pymolrc:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>antialias    </td><td>Set antialias to on to get smoother edges.               </td></tr>
<tr><td>fetchPath    </td><td>Set path for location to save fetched pdb files.         </td></tr>
<tr><td>lspymolrc    </td><td>Print list of active pymolrc files.                      </td></tr>
<tr><td>lspymolrcpy  </td><td>Print list of active pymolrc files.                      </td></tr>
<tr><td>setpath      </td><td>Set additional path for PyMOL to search on startup.      </td></tr>
<tr><td>sigDigits    </td><td>Set number of decimals places to show in distance labels.</td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='Selection'> <h3>Selection:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>duplicateObject     </td><td>Duplicate object. Create an object with the first argument using the selection which is the second argument.</td></tr>
<tr><td>extractPartObj      </td><td>Create a new object from part of an existing object.                                                        </td></tr>
<tr><td>hideSelection       </td><td>Turn off magenta squares on current selection.                                                              </td></tr>
<tr><td>ligandSelect        </td><td>Make selection of ligand atoms.                                                                             </td></tr>
<tr><td>selectAllBut        </td><td>Select all nitrogen atom in a selelction except from lysine.                                                </td></tr>
<tr><td>selectAtomsAround   </td><td>Select atoms within a radius around a ligand.                                                               </td></tr>
<tr><td>selectChain         </td><td>Select a chain.                                                                                             </td></tr>
<tr><td>selectElement       </td><td>Select atoms by element.                                                                                    </td></tr>
<tr><td>selectHelices       </td><td>Select atoms by alpha helices.                                                                              </td></tr>
<tr><td>selectLoops         </td><td>Select atoms by beta loops.                                                                                 </td></tr>
<tr><td>selectName          </td><td>Select atoms by name.                                                                                       </td></tr>
<tr><td>selectResi          </td><td>Select residues by a range of numbers.                                                                      </td></tr>
<tr><td>selectResidues      </td><td>Select residues by name.                                                                                    </td></tr>
<tr><td>selectResiduesAround</td><td>Select residues within a radius around a ligand.                                                            </td></tr>
<tr><td>selectStrands       </td><td>Select atoms by beta strands.                                                                               </td></tr>
<tr><td>undoSelection       </td><td>Undo a selection.                                                                                           </td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='Settings'> <h3>Settings:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>listSettings </td><td>Print to the screen the settings and their current parameter values. This is the more compact version.</td></tr>
<tr><td>listSettings2</td><td>Print to the screen the settings and their current parameter values.                                  </td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='Stereo'> <h3>Stereo:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>stereoDraw   </td><td>Stereo draw.                                  </td></tr>
<tr><td>stereoRay    </td><td>Stereo ray.                                   </td></tr>
<tr><td>stereokb     </td><td>Set keyboard shortcut by mapping F1 to stereo.</td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='Trajectories'> <h3>Trajectories:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>loadAmberTrajs</td><td>The amber trajectories have to be loaded into the same object.</td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='UnitCellDisplay'> <h3>Unit cell display:</h3> </a>
<table>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>addAxis                </td><td>Adds the function draw_axis(). Used to draw a symmetry axis, a ncs axis, or scale bar to a scene.                                                                                                                                                                                                                                                                                                              </td></tr>
<tr><td>addAxispy              </td><td>Adds the function draw_axis(). Could be useful for the adding a symmery axis, a ncs axis, or scale bar to a scene.                                                                                                                                                                                                                                                                                             </td></tr>
<tr><td>sc111                  </td><td>Display all symmetry mates in one unit cell. Uses supercell.py in $HOME/Scripts/PyMOLscripts/. Change to your path to supercell.py.                                                                                                                                                                                                                                                                            </td></tr>
<tr><td>sc112                  </td><td>Display all symmetry mates in two unit cells along the c axis. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                               </td></tr>
<tr><td>sc113                  </td><td>Display all symmetry mates in three unit cels along c. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                                       </td></tr>
<tr><td>sc114                  </td><td>Display all symmetry mates in four unit cells stacked long c-axis. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                           </td></tr>
<tr><td>sc121                  </td><td>Display all symmetry mates in two unit cells along the b axis. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                               </td></tr>
<tr><td>sc122                  </td><td>Display all symmetry mates in a 1 x 2 x 2 array of unit cells. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                               </td></tr>
<tr><td>sc131                  </td><td>Display all symmetry mates in three unit cells along b. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                                      </td></tr>
<tr><td>sc133                  </td><td>Display all symmetry mates in 1 x 3 x 3 array of unit cell. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                                  </td></tr>
<tr><td>sc141                  </td><td>Display all symmetry mates in four unit cells stacked long b-axis. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                           </td></tr>
<tr><td>sc144                  </td><td>Display all symmetry mates in in a 1 x 4 x 4 array. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                                          </td></tr>
<tr><td>sc211                  </td><td>Display all symmetry mates in two unit cell along a. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                                         </td></tr>
<tr><td>sc212                  </td><td>Display all symmetry mates in a 2 x 1 x 2 arrays of unit cells. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                              </td></tr>
<tr><td>sc221                  </td><td>Display all symmetry mates in 2 x 2 x 1 array of unit cells. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                                 </td></tr>
<tr><td>sc222                  </td><td>Run supercell script to generate three cells in all directions. This script was written by Thomas Holder.                                                                                                                                                                                                                                                                                                      </td></tr>
<tr><td>sc233                  </td><td>Display all symmetry mates in a 2 x 3 x 3 array of unit cells. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                               </td></tr>
<tr><td>sc311                  </td><td>Display all symmetry mates three three unit cells along a. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                                   </td></tr>
<tr><td>sc313                  </td><td>Display all symmetry mates in a 3 x 1 x 3 array of unit cells. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                               </td></tr>
<tr><td>sc323                  </td><td>Display all symmetry mates in a 3 x 2 x 3 array of unit cells. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                               </td></tr>
<tr><td>sc331                  </td><td>Display all symmetry mates in 3 x 3 x 1 array of unit cells. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                                 </td></tr>
<tr><td>sc332                  </td><td>Display all symmetry mates in 3 x 3 x 2 array of unit cells. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                                 </td></tr>
<tr><td>sc333                  </td><td>Display all symmetry mates in 3 x 3 x 3 array of unit cells. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                                 </td></tr>
<tr><td>sc411                  </td><td>Display all symmetry mates in four unit cells stacked long a-axis. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                           </td></tr>
<tr><td>sc414                  </td><td>Display all symmetry mates in a 4 x 1 x 4 array. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                                             </td></tr>
<tr><td>sc441                  </td><td>Display all symmetry mates in four unit cells stacked long a-axis. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                           </td></tr>
<tr><td>sc444                  </td><td>Display all symmetry mates in a 4 x 4 x4 array of unit cells. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                                </td></tr>
<tr><td>sc444                  </td><td>Display all symmetry mates in a 4 x 4 x4 array of unit cells. Uses supercell.py in $HOME/Scripts/PyMOLscripts/.                                                                                                                                                                                                                                                                                                </td></tr>
<tr><td>symexp                 </td><td>The code expands the asymmetric unit. It like the generate symmetry mates command but it provides more control over the prefix name of the symmetry mates and the addition of unique segment identifiers for each symmetry mate. The usage: symexp prefix, object, (selection), cutoff, segidFlag. The cutoff is in Angstroms. The segidFlag set to 1 will add unique segids. For related functions, see SC***.</td></tr>
<tr><td>unitCellEdgesColorBlack</td><td>Color unit cell edges black. The settings for controlling the unit cell color are hard to find.                                                                                                                                                                                                                                                                                                                </td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>


<a name='Workshop'> <h3>Workshop:</h3> </a>
<table class="table2">
<thead>
       <tr>
        <th WIDTH="80">Snippet Name </th>
	<th WIDTH="300">Description </th>
      </tr>  
</thead>
<tbody>
<tr><td>internalGUImode2</td><td>Make the background of the internal gui transparent to expand viewport.     </td></tr>
<tr><td>internalGUIwidth</td><td>Set the width of the internal gui. Set to 0 to make the internal gui vanish.</td></tr>
</tbody>
</table>
<a href=#categories>Return to list of snippet categories near top.</a>

</Body>
</HTML>
