cmd.do('"""Tab trigger                  Description')
cmd.do('---------------------------  --------------------------------------------------------------------------------------------------------------------------------------------------------------------')
cmd.do('antialias                    Set antialias to get smooth edges')
cmd.do('ao                           Ambient occlussion.')
cmd.do('aveB4resiX                   AveBResiX, prints the resiude number and the average bfactor.')
cmd.do('                             Uses reduce and lambda, builtin Python functional porgramming functions.')
cmd.do('                             Note that you need to convert the length of the list of Bfactors from an integer to a float before division into the sum.')
cmd.do('averageB                     Uses a regular list as opposed to PyMOL"s stored list. Edit the selection as needed.')
cmd.do('bs                           Ball and stick representation.')
cmd.do('bu                           Biological unit.')
cmd.do('carvedDensity                Carved electron density.')
cmd.do('cblind                       Color blindness.')
cmd.do('centerpi                     Center pi.')
cmd.do('coordinate                   Coordinate covalent bonds to metals and H-bonds friom RNA.')
cmd.do('cribbon                      Color ribbon H red, strand yellow, loop green.')
cmd.do('cspheres                     Colored spheres.')
cmd.do('discreteCartoonColoring      Turn on discrete colors between secondary structure elements.')
cmd.do('distance                     H-bond distances.')
cmd.do('doubleBond                   Valence bond.')
cmd.do('drawHbonds                   Draw H-bonds.')
cmd.do('duplicateObject              Duplicate object.')
cmd.do('ellipcol                     Set ellipsoid color.')
cmd.do('extractPartObj               Create a new object from part of an existing object.')
cmd.do('fasta                        Print Fasta from PDB file.')
cmd.do('fetch2FoFc                   Fetch 2FoFc map.')
cmd.do('fetchCIF                     Fetch cif file.')
cmd.do('fetchFoFc                    Fetch fofc map.')
cmd.do('fetchPath                    Set path for location to save fetched pdb files.')
cmd.do('filledRing                   Filled rings in nucleic acids.')
cmd.do('findHbonds                   Find hbonds around a residue.')
cmd.do('fog                          Blur the background atoms.')
cmd.do('getCoordinates               Get coordinates.')
cmd.do('hbond                        Hbond setup.')
cmd.do('hbonddash                    H-bond dashes.')
cmd.do('hideSelection                Trun off magenta squares on current selection.')
cmd.do('hidealtloc                   Hide alt loc.')
cmd.do('internalGUImode2             Makes the background of the internal gui transparent with the viewport extended into this region of the gui. This may be a useful mode for workshops.')
cmd.do('internalGUIwidth             Set the width of the internal gui.')
cmd.do('labelCAs                     Label the CA atoms with the Ala333 style format')
cmd.do('labelMainChain               Label the main chain atoms by resn,resi,atom name.')
cmd.do('labelResnResi                Label CA atom with residue name and residue number.')
cmd.do('labelSS                      Label SS.')
cmd.do('labelWatersHOH               Label waters HOH.')
cmd.do('labelWatersW                 Label waters W.')
cmd.do('loadPDBbs                    Load PDB ball-and-stick.')
cmd.do('loadPDBfile                  Load a pdb file in the current directory.')
cmd.do('loadPDBnb                    Load PDB nb spheres.')
cmd.do('lsSnips                      List all snips by tab trigger and description')
cmd.do('lspymolrc                    Print list of active pymolrc files.')
cmd.do('molscriptRibbon              Molscript ribbons.')
cmd.do('ms                           Measure surface area.')
cmd.do('oneLetter                    One letter amino acid.')
cmd.do('pearl                        The pearl effect is made with two spheres with the outer sphere being transparent.')
cmd.do('printBs                      Print the B-factors of a residue.')
cmd.do('printBs2digits               Print B-values for a residue with the B"s rounded off to two decimal places.')
cmd.do('printBspartB                 Print B factors of part B of a rsidues.')
cmd.do('printDoc                     Print document string of a function.')
cmd.do('printNameB4ResiX             Print name and b-factor for a rsidue. You can change this to a named selection or a residue range ( e.g., resi 133:155). Use the noH variant if H atoms are present.')
cmd.do('printResiResnNameB4ResiX     Print resn, resi, atom name, and b-factor.')
cmd.do('printResiResnNameB4ResiXNoH  Print name and b-factor for a rsidue. You can change this to a named selection or a residue range ( e.g., resi 133:155). The noH variant.')
cmd.do('pseudolabel                  Position label with pseudoatom.')
cmd.do('puttyCartoon                 Create a putty cartoon.')
cmd.do('ringMode                     Set the ring mode to a value between 0  and 6 in cartoons of nucleic acids.')
cmd.do('rmwater                      Remove waters from molecular object.')
cmd.do('rotate                       Rotate about axis.')
cmd.do('rv                           Return settings in rounded format.')
cmd.do('savePNG                      Save a png file of current scene to the current directory.')
cmd.do('saxsEnvelope                 Display SAXS envelope')
cmd.do('sc111                        Display all symmetry mates in one unit cell. Asumes supercell.py (sse PyMOL Wiki) is in $HOME/Scripts/PyMOLscripts/.')
cmd.do('sc222                        Run Tom Holder"s supercell script to generate three cells in all directions.')
cmd.do('scaleRadiusColor             Scale the radius and color of atoms as spheres by property in the B-value column.')
cmd.do('selectAllBut                 Select all nitrogen atom in a selelction except from lysine.')
cmd.do('selectAtomsAround            Select atoms within a radius around a ligand.')
cmd.do('selectChain                  Select a chain.')
cmd.do('selectElement                Select atoms by elemnt.')
cmd.do('selectHelices                Select atoms by alpha helices.')
cmd.do('selectLoops                  Select atoms by beta loops.')
cmd.do('selectName                   Select atoms by name.')
cmd.do('selectResi                   Select residues by a range of numbers.')
cmd.do('selectResidues               Select residues by name.')
cmd.do('selectResiduesAround         Select residues within a radius around a ligand.')
cmd.do('selectStrands                Select atoms by beta strands.')
cmd.do('setcolor                     Set color name to a RGB code.')
cmd.do('setpath                      Set additional path for PyMOL to search on startup')
cmd.do('sidehChainHelper             In cartoons, hide the backbone atoms of selected residues when showing then as sticks.')
cmd.do('sigDigits                    Set number of decimals places to show in distance labels.')
cmd.do('sigang                       Set angle labels to display 2 decimals places')
cmd.do('sigdist                      set distance labels to display 2 decimals')
cmd.do('solventRadius                Set radius of ball used to make solvent accessible surface.')
cmd.do('spng                         Save png flle with timestamp')
cmd.do('spse                         Save pse flle with timestamp')
cmd.do('stack                        Base-stacking figure.')
cmd.do('stereoDraw                   Stereo draw.')
cmd.do('stereoRay                    Stereo ray.')
cmd.do('threeMaps                    Three electron density.')
cmd.do('turnAboutAxis                Turn about axis.')
cmd.do('undoSelection                Undo a selection.')
cmd.do('volumeRamp                   Volume ramp.')
cmd.do('writeCommandReference2HTML   Write the command reference to html file in the present working directory."""')
